### ISSUE TITLE
dropdown options in repeater field doesn't update properly when depending on another field

### ISSUE THREAD

**Issue by fresswolf** [octobercms/october#1891]
Let's assume the following configuration
```yaml
fields:
    one:
        label: One
    another:
        label: Tickets
        type: repeater
        dependsOn: one
        form:
            fields:
                two:
                    label: Two
                    type: dropdown
                    dependsOn: one
```

and further some dropdown options depending on the first field

```php
public function getTwoOptions()
    {
        if ($this->one) {
            return ['my new option: ' . $this->one];
        }
        return ['nothing'];
    }
```

Now, when I add one item and then change the value in "one", the dropdown options for "two" are refreshed as expected. However, if I first edit "one" and then add an item in the repeater, the options inside the repeater have not been refreshed

**Comment by daftspunk** [comment]
This is a complex case. Please submit a Pull  #Request to the [test plugin for investigation](https://github.com/daftspunk/oc-test-plugin) that demonstrates the issue. Don't forget to include step by step instructions how how to replicate it *using the test plugin*.

**Comment by daftspunk** [comment]
Fixed by: #2235
Our hero: @acasar

![Up top](https://dl.dropboxusercontent.com/u/5381773/lel/up-top.jpg)
